### PR TITLE
#35: Option to skip waiting for UDF container

### LINKED ISSUE #35: Option to skip waiting for UDF container
## Situation

If the test conducted with the Exasol-Testcontainer do not include User Defined Functions (UDFs), the tester can skip waiting for the UDF container to become ready and save some time in the test setup.

## Acceptance Criteria

1. User can configure the test container so that waiting for the UDF container to be ready can be skipped.

### LINKED ISSUE #36: NullPointerException in getJdbcConnection
When the container has not been started yet, a call to `getJdbcUrl()` ends up like this:

```
java.lang.NullPointerException
	at org.testcontainers.DockerClientFactory.dockerHostIpAddress(DockerClientFactory.java:240)
	at org.testcontainers.containers.ContainerState.getContainerIpAddress(ContainerState.java:29)
	at com.exasol.containers.ExasolContainer.getJdbcUrl(ExasolContainer.java:80)

```

Any other Exception or NULL would be fine :)

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #35.
Closes #36 (NULL pointer exception in `getJdbcUrl()` as this is based on `testcontainers` 1.12.5 which fixes a regression (see https://github.com/testcontainers/testcontainers-java/pull/2119).

Task-Url: http://github.com/exasol/exasol-testcontainers/issues/35

### CODE REVIEW

File: doc/design.md [hunk 1]
```diff
@@ -148,16 +159,20 @@ Covers:
...
-`dsn~exaloader-in-common-docker-network~1`
+### IP Address in Common Docker Network
+`dsn~ip-address-in-common-docker-network~1`
+
+ETC reports the docker-network internal IP address.
+
+Rationale:
 
-ETC allows running two Exasol containers in the same docker network so that the ExaLoader is able to execute `IMPORT` statements that import data from one database into the other.
+This is the main requirement for any Exasol service to be able to reach another service in the same Docker network. Hostname resolution via Docker network alias is not supported by `docker-db` yet.
```
Comment (morazow) [inline doc/design.md]:
```suggestion
This is the main requirement for any Exasol service to be able to reach another service in the same Docker network. Hostname resolution via Docker network alias is not supported by [`exasol/docker-db`](https://github.com/exasol/docker-db) yet.
```

File: src/main/java/com/exasol/containers/ExasolContainer.java [hunk 2]
```diff
@@ -145,6 +147,12 @@ public T withPassword(final String password) {
         return self();
     }
 
+    // [impl->dsn~defining-required-optional-service~1]
```
Comment (morazow) [inline src/main/java/com/exasol/containers/ExasolContainer.java]: Maybe javadoc is required here for public method. Not sure if sonar already complained about it.

File: src/main/java/com/exasol/containers/ExasolContainer.java [hunk 3]
```diff
@@ -270,4 +278,14 @@ public String getDockerNetworkInternalIpAddress() {
         }
         return "127.0.0.1";
     }
+
+    /**
+     * Check if an service is ready to be used.
```
Comment (morazow) [inline src/main/java/com/exasol/containers/ExasolContainer.java]:
```suggestion
     * Check if a service is ready to be used.
```

File: src/main/java/com/exasol/containers/ExasolContainer.java (hunk 2 above)
Comment (redcatbear) [reply src/main/java/com/exasol/containers/ExasolContainer.java]: Fixed.

**Pull request merged**